JenkinsfileDeploy

```diff
@@ -0,0 +1,42 @@
+def tag = env.TAG_NAME
+def version = tag.substring(1)      // Remove the first character of the tag which is 'v'
+def machines = "[dom daint TSAtds]" // To be included in the PR title
+def reviewersPR = "teojgo,vkarak"   // The Github usernames requested for review
+
+stage('Deploy Tests') {
+    node('dom') {
+        dir('private_tests') {
+            sh(returnStatus: true,
+               script: """#!/bin/bash -l
+                          git clone git@github.com:eth-cscs/reframe-tests.git
+                          cd reframe-tests
+                          git tag -a -m "ReFrame private tests ${version} release" -f ${tag}
+                          git archive --format=tar.gz --prefix=reframe-tests-${version}/ ${tag} -o reframe-tests-${version}.tar.gz
+                          git push origin -f ${tag}
+                          cp reframe-tests-${version}.tar.gz /apps/common/easybuild/sources/r/reframe
+                          """)
+            deleteDir()
+        }
+    }
+}
+
+stage('Production PR') {
+    node('dom') {
+        dir('production_repo') {
+            sh(returnStatus: true,
+               script: """#!/bin/bash -l
+                          module load daint-gpu
+                          module load hub
+                          git clone git@github.com:eth-cscs/production.git
+                          cd production/easybuild/easyconfigs/r/reframe
+                          git checkout -b reframe/${version}
+                          sed 's/3.0-dev5/${version}/g' reframe-3.0-dev5.eb > reframe-${version}.eb
```

Comment:
I wouldn't hardcode the base version here, since we will need to change it from time to time.
Also why are you doing a sed across the whole file and don't just change the version? If we have the version hardcoded elsewhere in the easyconfig, it's just badly written.